### ISSUE TITLE
Reject promise with NotSupportedError on unsupported keysystem in 'HDCP Policy Check' proposal (moved from wicg/media-capabilities)

### ISSUE THREAD

**Issue by mounirlamouri** [WICG/hdcp-detection#2]
Original issue: https://github.com/WICG/media-capabilities/issues/56

/CC @xhwang-chromium @JamesWCCheng

**Comment by joeyparrish** [comment]
@JamesWCCheng, this does not make sense because the new method is on `MediaKeys`.  You can't get a `MediaKeys` instance if the key system is unsupported, so this error would have been reached during `requestMediaKeySystemAccess`.

(Cross-posted to the original discussion as well.)

**Comment by xhwang-chromium** [comment]
Reading through the old bug, I think "unsupported key system" here means key systems that do not implement/support this "hdcp detection" API, not a key system not supported by the UA. For example, Clear Key key system doesn't do anything about HDCP, and can potentially declare that "hdcp detection" API is not supported.

I see mounirlamouri@'s point that the CDM should "just do what it would do with a real license with certain limitation/requirement (e.g. HDCP reqirement)". The current explainer doc also says:
> The new API will allow application developers to query the status of a hypothetical key associated with an HDCP policy...

However, in the example of Clear Key key system, it never expects a license with HDCP requirement. As such, a typically Clear Key CDM implementation wouldn't have code to check HDCP status to determine whether a "hypothetical key associated with an HDCP policy" would be usable or not. Therefore, it's hard and unnecessary to ask all CDMs to implement this API, and I think it makes more sense to have these CDMs declare "HDCP detection" API is not supported.

In summary, I think it makes sense that if a MediaKeys (CDM) doesn't not support this policy check that it should reject the promise with NotSupportedError. This is actually how things are [implemented in Blink/Chromium](https://cs.chromium.org/chromium/src/media/base/content_decryption_module.cc?rcl=b856f58e77f6d151515e0cfba263cd9d4515be08&l=19).

If agreed, we should update the explainer doc to reflect this.

**Comment by joeyparrish** [comment]
Okay, that makes sense to me.

**Comment by xhwang-chromium** [comment]
joeyparrish: I am a bit confused with all the labels :) Just want to make sure this is included when you update the spec.
